system/storage/copy.go

```diff
@@ -34,11 +34,12 @@ func (s *service) Copy(context *endly.Context, request *CopyRequest) (*CopyRespo
 func (s *service) copy(context *endly.Context, request *CopyRequest, response *CopyResponse) error {
 	var udfModifier option.Modifier
 	if request.Udf != "" {
+		var ok bool
 		UDF, err := udf.TransformWithUDF(context, request.Udf, "", nil)
 		if err != nil {
-			return errors.Wrapf(err, "failed to Copy")
+			return errors.Wrapf(err, "failed to get udf: %v", request.Udf)
 		}
-		udfModifier, ok := UDF.(option.Modifier)
+		udfModifier, ok = UDF.(option.Modifier)
 		if !ok {
 			return fmt.Errorf("udf %v does not implement %T", UDF, udfModifier)
```

Comment:
consistency either user error.Wrap or fmt.Errorf

Reply:
since this is integration refactoring with any new line of code I would use errors.Wrapf, 
I have not looked into fmt.Errorf past usage, good spt

Reply:
I guess, fmt.Errorf still makes sense to use when there is no previous error to preserve